### PR TITLE
CI: Update pipelines config to trigger on PRs

### LINKED ISSUE #32705: CI: azure pipelines are not in PRS
cc @pandas-dev/pandas-core

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] closes #32705

### CODE REVIEW

**Comment by WillAyd** [comment]
Thanks for this! Not sure what is up with the failure - just restarted let's see...

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # Adapted from https://github.com/numba/numba/blob/master/azure-pipelines.yml
+trigger:
+- master
+
+pr:
+- master
```
Comment (WillAyd) [inline azure-pipelines.yml]: Do you know of a way to just do this for all PRs? I have a hunch that this wouldn't run now for backports but didn't see anything in the docs:

https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=example%2Cparameter-schema#pr-trigger

Comment (WillAyd) [reply azure-pipelines.yml]: Maybe just need to specify as *:

https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml#wildcards

Though I think it's a bug that we need this in the first place given this in the docs:

> A pull request trigger specifies which branches cause a pull request build to run. If you specify no pull request trigger, pull requests to any branch trigger a build

**Comment by jreback** [comment]
hmm, is the 32-bit build borked? we can just merge this and followup if needed later.

**Comment by WillAyd** [comment]
Yea I guess need a follow up to get rid of the 32 bit one though; looks to be failing

**Comment by jreback** [comment]
@TomAugspurger marking for 1.0.3 if needed

**Comment by simonjayhawkins** [comment]
This is marked as still needs manual backport. looking at the code changes there is reference to master. What needs to be done here?

**Comment by datapythonista** [comment]
Not sure, I guess if we want to run the CI on other branches, the name of the branch (`master` in the diff) should be updated for each branch.

For context, without these changes, pipelines wasn't running.

**Comment by simonjayhawkins** [comment]
> For context, without these changes, pipelines wasn't running.

the azure jobs are running on 1.0.x, its the code checks that are not running

**Comment by simonjayhawkins** [comment]
i've removed the needs manual backport, but will probably resurface after 1.1.0 is released.

**Comment by simonjayhawkins** [comment]
> i've removed the needs manual backport, but will probably resurface after 1.1.0 is released.

cc @TomAugspurger

**Comment by simonjayhawkins** [comment]
I think needs to be changed in master so that when the 1.1.x branch is created, backport PRs will be checked

**Pull request merged**